From dc7049b6fdd417c277a81aaf739fdb04456685a6 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Fri, 14 Dec 2012 13:05:34 +0200
Subject: [PATCH 3/4] OF: Compile Device Tree sources with resolve option

If the resolve config option has been enabled compile
the device tree sources with the appropriate flag to generate
the required for resolution __symbols__ node.

Note that both arch/arm/Makefile & arch/arm/boot/Makefile need
the DTC_FLAGS definition, because when issuing make dtbs the
boot Makefile doesn't seem to pick it.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 arch/arm/Makefile      | 6 +++++-
 arch/arm/boot/Makefile | 4 ++++
 2 files changed, 9 insertions(+), 1 deletion(-)

Index: KERNEL/arch/arm/Makefile
===================================================================
--- KERNEL.orig/arch/arm/Makefile	2014-01-05 16:40:53.320998669 +0100
+++ KERNEL/arch/arm/Makefile	2014-01-05 16:44:56.474577484 +0100
@@ -286,6 +286,10 @@
 KBUILD_DTBS := dtbs
 endif
 
+ifeq ($(CONFIG_OF_RESOLVE),y)
+DTC_FLAGS += -@
+endif
+
 all:	$(KBUILD_IMAGE) $(KBUILD_DTBS)
 
 boot := arch/arm/boot
@@ -312,7 +316,7 @@
 
 PHONY += dtbs
 dtbs: scripts
-	$(Q)$(MAKE) $(build)=$(boot)/dts MACHINE=$(MACHINE) dtbs
+	$(Q)$(MAKE) $(build)=$(boot)/dts MACHINE=$(MACHINE) DTC_FLAGS=$(DTC_FLAGS) dtbs
 
 # We use MRPROPER_FILES and CLEAN_FILES now
 archclean:
Index: KERNEL/arch/arm/boot/Makefile
===================================================================
--- KERNEL.orig/arch/arm/boot/Makefile	2014-01-05 16:40:53.321998560 +0100
+++ KERNEL/arch/arm/boot/Makefile	2014-01-05 16:44:56.474577484 +0100
@@ -57,6 +57,10 @@
 
 endif
 
+ifeq ($(CONFIG_OF_RESOLVE),y)
+DTC_FLAGS += -@
+endif
+
 ifneq ($(LOADADDR),)
   UIMAGE_LOADADDR=$(LOADADDR)
 else
